parlai/core/worlds.py

```diff
@@ -440,6 +440,19 @@ def __init__(self, opt, agents=None, shared=None, default_world=None):
         self.new_world = True
         self.parleys = -1
         self.random = opt.get('datatype', None) == 'train'
+        # Make multi-task task probabilities.
+        # These come from the 'taskweight' param (if it exists)
+        #  in the teacher from each world (assumed to be agent 0).
+        self.cum_task_weights = [1] * len(self.worlds)
+        self.task_choices = range(len(self.worlds))
+        sum = 0
+        for i, w in enumerate(self.worlds):
+            agents = w.get_agents()
+            weight = 1
+            if len(agents) > 0:
+                weight = float(agents[0].opt.get('taskweight', 1))
```

Comment:
let's use just `opt`, instead of `agents[0].opt`

Looks like you need to add the command line param to train_model.py. Worth noting that @alexholdenmiller recently added the `--type='floats'` param that pre-parses the options into a tuple of floats.

Reply:
ok, changed agents[0].opt that seems to work..
but it's actually an agent specific parameter though, not a global paramater of the opts (per task) the way i've done it.. but i could do it other ways.. so not really to do with train_model

Reply:
fixed